src/viewProviders/resources.ts

```diff
@@ -120,8 +122,12 @@ export class ResourceViewProvider implements vscode.TreeDataProvider<ResourceVie
     return ResourceViewProvider.instance;
   }
 
-  getTreeItem(element: ResourceViewProviderData): vscode.TreeItem {
+  async getTreeItem(element: ResourceViewProviderData): Promise<vscode.TreeItem> {
     if (element instanceof Environment) {
+      if (isDirect(element)) {
+        // update contextValues for all known direct environments, not just the one that was updated
+        await this.updateEnvironmentContextValues();
```

Comment:
This is the meat of the new behavior; any time we use `_onDidChangeTreeData.fire(item)`, this is called (see https://github.com/confluentinc/vscode/pull/780).